_includes/types/uri.md

```diff
@@ -16,6 +16,13 @@ We also maintain a node module to parse a string into `scheme`, `authority`, `pa
 
 Many of the interfaces contain fields that correspond to the URI of a document. For clarity, the type of such a field is declared as a `DocumentUri`. Over the wire, it will still be transferred as a string, but this guarantees that the contents of that string can be parsed as a valid URI.
 
+Care should be taken to handle encoded in URIs. Some clients (such as VS Code) may encode colons in drive letters while others do not. The following URIs should be considered equivalent and are both valid for clients and servers to use:
```

Comment:
> However clients should be consistent in how they encode characters in URIs.

I don't like the word _should_ 😁

I've pushed some tweaks - how's this? (I explicitly noted that one party should not assume the other party will use the same encoding, because I think it has to be that way - perhaps I should have noted the same about drive letter casing?)

Reply:
What could also be problematic and where care should be taken is when clients send symlinked `file` URIs. Or dealing with case-insensitive filenames. I would argue a server cannot rely on the string-representation of `file` URIs.

Reply:
Yep, there are definitely more edge cases than this, but this one is simple (it's just encoding/decoding) and already caused me a problem. It's much less clear to me what the spec should say about symlinks or casing so I think they could be handled in their own PRs (assuming some rules can be agreed!).